### PR TITLE
sql: add help text for show zone configuration

### LINKED ISSUE #58689: sql: add help for SHOW ZONE CONFIGURATION in SQL shell
**Is your feature request related to a problem? Please describe.**
Running `\h SHOW` in the SQL shell doesn't list SHOW ZONE CONFIGURATIONS and there is no help entry for it in
`pkg/sql/parser/sql.y`.
```
demo@127.0.0.1:26257/movr> \h SHOW
Command:     SHOW
Syntax:
SHOW BACKUP, SHOW CLUSTER SETTING, SHOW COLUMNS, SHOW CONSTRAINTS,
SHOW CREATE, SHOW DATABASES, SHOW ENUMS, SHOW HISTOGRAM, SHOW INDEXES, SHOW
PARTITIONS, SHOW JOBS, SHOW STATEMENTS, SHOW RANGE, SHOW RANGES, SHOW REGIONS, SHOW SURVIVAL GOAL,
SHOW ROLES, SHOW SCHEMAS, SHOW SEQUENCES, SHOW SESSION, SHOW SESSIONS,
SHOW STATISTICS, SHOW SYNTAX, SHOW TABLES, SHOW TRACE, SHOW TRANSACTION,
SHOW TRANSACTIONS, SHOW TYPES, SHOW USERS, SHOW LAST QUERY STATISTICS, SHOW SCHEDULES,
SHOW LOCALITY
```

**Describe the solution you'd like**
A help entry should be added to `pkg/sql/parser/sql.y` for SHOW ZONE CONFIGURATIONS.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by tharun208**
Release justification: low risk, high benefit changes to existing functionality

This PR adds show zone command help text in sql shell

Resolves #58689

Release note: None

Signed-off-by: Tharun <rajendrantharun@live.com>

### CODE REVIEW

**Comment by tharun208** [comment]
bors r=tbg

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -5226,6 +5226,12 @@ show_roles_stmt:
   }
 | SHOW ROLES error // SHOW HELP: SHOW ROLES
 
+// %Help: SHOW ZONE CONFIGURATION - display current zone configuration
+// %Category: Cfg
+// %Text: SHOW ZONE CONFIGURATION FOR [ RANGE | DATABASE | TABLE | INDEX ] <name>
+// SHOW ZONE CONFIGURATION FOR PARTITION OF [ INDEX | TABLE ] <name>
```
Comment (andyyang890) [inline pkg/sql/parser/sql.y:5232]: Please replace `FOR` with `FROM`.

**Review by andyyang890 — COMMENTED** [review]
Thanks for this! Left a comment on the help text.

Could you please also make the following changes to the commit message and PR comment:

1. Change the release note to "Release note: None".
2. Change the release justification to "Release justification: low risk, high benefit change to existing functionality".

File: pkg/sql/parser/sql.y (hunk 1 above)
Comment (tharun208) [reply pkg/sql/parser/sql.y:5232]: @andyyang890 , But , I am seeing `FOR` in the docs - https://www.cockroachlabs.com/docs/stable/show-zone-configurations.html

Comment (andyyang890) [reply pkg/sql/parser/sql.y:5232]: It was changed in https://github.com/cockroachdb/cockroach/pull/59410

Comment (tharun208) [reply pkg/sql/parser/sql.y]: done, thanks

**Comment by tharun208** [comment]
@andyyang890, I see a failure in the ci, can you retrigger it again? thanks for the help in advance.

**Review by andyyang890 — APPROVED** [review]
LGTM! Adding @rafiss for TL approval

**Review by rafiss — APPROVED** [review]
thanks for your contribution!

bors r+

**Comment by tharun208** [comment]
@andyyang890 @rafiss , I think we also need a doc to change for this as the docs show `FOR` instead of `FROM`.

**Comment by andyyang890** [comment]
@tharun208 There is a release note on the commit making the change from `FOR` to `FROM` so it will be documented in the 21.1 release.

**Comment by tharun208** [comment]
okay thanks

**Pull request merged**